crates/oxc_transformer/src/typescript/namespace.rs

```diff
@@ -150,9 +151,15 @@ impl<'a> TypeScript<'a> {
             &ctx.generate_uid_in_current_scope(real_name, SymbolFlags::FunctionScopedVariable),
         );
 
+        // TODO: This is workaround for now.
```

Comment:
Personally I think this can be a permanent solution rather than temporary.

The advantage of re-using the existing scope is that then all child scopes have the correct parent, and bindings within the block are bound in correct scope already. So we don't have to do any more work updating the scope tree the way we would if we created a new scope.

Reply:
I agreed